File: 1877. Minimize Maximum Pair Sum in Array.cpp
class Solution {
public:
    int minPairSum(vector<int>& nums) 
    {
        sort(nums.begin(), nums.end());
        int ans = 0;
        int n = nums.size(), i = 0, j = n-1;
        while(i < j)
            ans = max(ans, nums[i++] + nums[j--]);
        return ans;
    }
};

File: Sorting/1877. Minimize Maximum Pair Sum in Array.py
class Solution:
    def minPairSum(self, nums: List[int]) -> int:
        nums.sort()
        return max([nums[i] + nums[len(nums) - i - 1] for i in range(len(nums) // 2)])

class Solution:
    def minPairSum(self, nums: List[int]) -> int:
        nums.sort()

        ans = 2
        i, j, = 0, len(nums) - 1
        while i < j:
            ans = max(ans, nums[i] + nums[j])
            i += 1
            j -= 1
        return ans